### PR TITLE
Add check to test is WAF is attached to Cloudfront distribution #146

### PR DESCRIPTION
**Pull request by JamesWoolfenden**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
This is an attempt to resolve "Add new check: CloudFront Distribution should have WAF enabled #146" for Terraform. Im trying to use you new HCL2 format so hopefully this conforms?
 I tested and build and installed the package and it works with my module https://github.com/JamesWoolfenden/terraform-aws-cloudfront-s3.

### CODE REVIEW

File: checkov/terraform/checks/resource/aws/WAFEnabled.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+from checkov.common.models.consts import ANY_VALUE
+from checkov.terraform.checks.resource.base_resource_value_check import BaseResourceValueCheck
+
+
+class WAFEnabled(BaseResourceValueCheck):
+    def __init__(self):
+        name = "CloudFront Distribution should have WAF enabled"
+        id = "CKV_AWS_68"
+        supported_resources = ['aws_cloudfront_distribution']
+        categories = [CheckCategories.ENCRYPTION]
```
Comment (schosterbarak) [inline checkov/terraform/checks/resource/aws/WAFEnabled.py:11]: I think WAF deserves a new category. WDYT on CheckCategories.APPLICATION_SECURITY ?

File: checkov/terraform/checks/resource/__init__.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
 from checkov.terraform.checks.resource.aws import *
 from checkov.terraform.checks.resource.gcp import *
 from checkov.terraform.checks.resource.azure import *
+from checkov.terraform.checks.resource.github import *
```
Comment (schosterbarak) [inline checkov/terraform/checks/resource/__init__.py:4]: sweet!

Comment (JamesWoolfenden) [reply checkov/terraform/checks/resource/__init__.py:4]: i got a little carried away here, looks like this PR also has the new provider and test for github

File: checkov/terraform/checks/resource/aws/WAFEnabled.py (hunk 1 above)
Comment (JamesWoolfenden) [reply checkov/terraform/checks/resource/aws/WAFEnabled.py]: Cool, added.

File: checkov/terraform/checks/resource/__init__.py (hunk 2 above)
Comment (schosterbarak) [reply checkov/terraform/checks/resource/__init__.py:4]: np. looks awesome. would you like to merge those together? 
If you feel like the github is ready, we can merge those.

Comment (JamesWoolfenden) [reply checkov/terraform/checks/resource/__init__.py:4]: this was supposed to be a separate PR but its all the code to add support for the github provider and its first check. Hope thats ok or i can split to another PR. Got a bit keen.

**Pull request merged**